catalog/plugin/catalog-plugin-metacard-validation/src/main/java/ddf/catalog/metacard/validation/MetacardValidityCheckerPlugin.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+
+package ddf.catalog.metacard.validation;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.filter.FilterAdapter;
+import ddf.catalog.filter.FilterBuilder;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.impl.QueryImpl;
+import ddf.catalog.operation.impl.QueryRequestImpl;
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.plugin.PreQueryPlugin;
+import ddf.catalog.plugin.StopProcessingException;
+import ddf.catalog.source.UnsupportedQueryException;
+
+public class MetacardValidityCheckerPlugin implements PreQueryPlugin {
+
+    private static final Logger LOGGER = LoggerFactory
+            .getLogger(MetacardValidityCheckerPlugin.class);
+
+    protected static FilterBuilder filterBuilder;
+
+    protected static FilterAdapter filterAdapter;
+
+    public MetacardValidityCheckerPlugin(FilterBuilder filterBuilder, FilterAdapter filterAdapter) {
+        MetacardValidityCheckerPlugin.filterBuilder = filterBuilder;
+        MetacardValidityCheckerPlugin.filterAdapter = filterAdapter;
+    }
+
+    @Override
+    public QueryRequest process(QueryRequest input)
+            throws PluginExecutionException, StopProcessingException {
+        QueryImpl query;
+        try {
+            if (!filterAdapter.adapt(input.getQuery(), new ValidityParamDelegate())) {
+                query = new QueryImpl(filterBuilder.allOf(input.getQuery(),
+                        filterBuilder.attribute(MetacardValidityMarkerPlugin.VALIDATION_ERRORS).is()
+                                .empty(),
+                        filterBuilder.attribute(MetacardValidityMarkerPlugin.VALIDATION_WARNINGS)
+                                .is().empty()));
+            } else {
+                query = new QueryImpl(input.getQuery());
+            }
+        } catch (UnsupportedQueryException e) {
+            LOGGER.debug("This attribute filter is not supported by ValidityParamDelegate.", e);
+            throw new StopProcessingException(e.getMessage());
+        }
+        return new QueryRequestImpl(query);
```

Comment:
You are currently stripping all of the other QueryRequest data from `input`.  You need to use the `QueryRequestImpl(Query query, boolean isEnterprise, Collection<String> siteNames, Map<String, Serializable> properties)` constructor to copy all of the other fields from `input`.